Bottom: 5432ac67408f3825825d4bde2a0f8df2f77fa276
Top:    872a8c1a3db699bc068ed21722e24955b2987c5e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 17:49:40 +0000

Refresh of hid-gtk-gui-drc-window-c-refac

---

diff --git a/src/hid/gtk/gui-drc-window.c b/src/hid/gtk/gui-drc-window.c
index 5d4bb35..70b3a5d 100644
--- a/src/hid/gtk/gui-drc-window.c
+++ b/src/hid/gtk/gui-drc-window.c
@@ -92,18 +92,6 @@ enum {
 
 
 static void
-unset_found_flags (int AndDraw)
-{
-  bool changed = ResetConnections (true, FOUNDFLAG);
-
-  if (changed && AndDraw)
-    {
-      IncrementUndoSerialNumber ();
-      Draw ();
-    }
-}
-
-static void
 selection_changed_cb (GtkTreeSelection *selection, gpointer user_data)
 {
   GtkTreeModel *model;
@@ -113,7 +101,11 @@ selection_changed_cb (GtkTreeSelection *selection, gpointer user_data)
 
   if (!gtk_tree_selection_get_selected (selection, &model, &iter))
     {
-      unset_found_flags (true);
+      if (ResetConnections (true, FOUNDFLAG))
+        {
+          IncrementUndoSerialNumber ();
+          Draw ();
+        }
       return;
     }
 
@@ -123,7 +115,7 @@ selection_changed_cb (GtkTreeSelection *selection, gpointer user_data)
 
   gtk_tree_model_get (model, &iter, DRC_VIOLATION_OBJ_COL, &violation, -1);
 
-  unset_found_flags (false);
+  ResetConnections (true, FOUNDFLAG);
 
   if (violation == NULL)
     return;
